lib/puppet/util/execution.rb

```diff
@@ -98,11 +98,11 @@ def self.exitstatus
   # @raise [exception] under same conditions as {execute}, but raises the given `exception` with the output as argument
   # @return (see execute)
   # @api public
+  # @deprecated
```

Comment:
Do you also want to deprecate the provider wrappers?
https://github.com/puppetlabs/puppet/blob/ebd96213cab43bb2a8071b7ac0206c3ed0be8e58/lib/puppet/provider.rb#L120-L129

In our code, looks like RPM provider (and its derivatives) are the only callers - https://github.com/puppetlabs/puppet/blob/8a6dd1b9034e0d7e403c524fe74b297825b6a93a/lib/puppet/provider/package/rpm.rb#L135